pkg/controller/core.oam.dev/v1alpha2/applicationrollout/applicationrollout_controller.go

```diff
@@ -94,29 +94,39 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (res reconcile.Result, retErr e
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 	klog.InfoS("Start to reconcile ", "appRollout", klog.KObj(&appRollout))
-	if len(appRollout.Status.RollingState) == 0 {
-		appRollout.Status.ResetStatus()
-	}
-	targetAppRevisionName := appRollout.Spec.TargetAppRevisionName
-	sourceAppRevisionName := appRollout.Spec.SourceAppRevisionName
 
 	// handle app Finalizer
 	doneReconcile, res, retErr := r.handleFinalizer(ctx, &appRollout)
```

Comment:
does the finalizer not need to be treated specially? Will the appRollout get a deletion timestamp?

Reply:
Yes, finalizer will be set in appRollout.